docs/developer/extending/apps/synchronous-webhooks.mdx

```diff
@@ -0,0 +1,200 @@
+[
+   {
+      "id":"credit-card",
+      "name":"Credit Card",
+      "currencies":[
+         "EUR"
+      ],
+      "config":[]
+   },
+   {
+      "id":"apple-pay",
+      "name":"Apple Pay",
+      "currencies":[
+         "EUR",
+         "USD"
+      ],
+      "config":[
+         {
+            "field":"apple-pay-config-field",
+            "value":"value"
+         }
+      ]
+   }
+]
+```
+
+Each item on the list should have the following fields:
+- `id` - ID of a payment method; must be unique within your app.
+- `name` - the name of the payment method.
+- `currencies` - list of currencies supported by this payment method.
+- `config` - optional configuration, such as additional payment data that may be needed to proceed with the payment in the frontend.
+
+
+### Register webhooks for payment actions
+
+To subscribe to particular payment events (such as capture or authorization), you can have multiple webhooks with separate URLs for each action or a single webhook that handles all events (each webhook has an `X-Saleor-Event` header that contains the type of event).
+
+List of payment events: 
+
+- `PAYMENT_AUTHORIZE`
+- `PAYMENT_CAPTURE`
+- `PAYMENT_CONFIRM`
+- `PAYMENT_PROCESS`
+- `PAYMENT_REFUND`
+- `PAYMENT_VOID`
+
+:::note
+If an app has more than one webhook that listens to the same payment event, Saleor will always pick the first webhook from the list.
+:::
+
+See the example payload sent from Saleor for these payment events: [Payment actions payload](/developer/extending/apps/sample-webhook-payloads.mdx#payment-actions-payload).
+
+For payments that require confirmation (such as 3D Secure) Saleor triggers two events:
+- `PAYMENT_PROCESS` - when the `checkoutComplete` mutation is run for the first time, payment is created and waiting for the user's confirmation.
+- `PAYMENT_CONFIRM` - when the `checkoutComplete` mutation is run for the second time, payment is confirmed.
+
+For payments without a confirmation step, Saleor triggers only the `PAYMENT_PROCESS` event.
+
+#### Expected response
+
+Below is the list of fields that webhook can return in the JSON response for a payment action. All fields are optional:
+
+- `action_required` - (boolean) determines if additional action (confirmation) is required.
+- `action_required_data` - (object) additional data that may be needed to confirm the payment, for example, URL to redirect the user to the confirmation page in the frontend.
+- `amount` - (number) transaction amount; use this field only if the charged amount is different than checkout's total.
+- `customer_id` - (string) gateway's customer ID.
+- `error` - (string) error message; if provided, the transaction will be marked as failed in Saleor and payment will have to be retried. Use this field if you need to return any errors.
+- `payment_method` - (object) information about payment method chosen by the user.
+- `psp_reference` - (string) reference which can be used to find a payment object in Saleor, once the order is created.
```

Comment:
`reference which can be used to find a payment object in Saleor` there is no any API which you could use to find payment object. AFAIK we use psp_reference to search order with such reference.